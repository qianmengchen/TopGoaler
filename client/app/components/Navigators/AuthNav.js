import React, { Component } from 'React';
import { SignIn } from '../SignIn/index';
import { SignUp } from '../SignUp/index';
import { createSwitchNavigator } from 'react-navigation';

import TabNav from './TabNav';
import { connect } from 'react-redux';

const SignNav = createSwitchNavigator(
  {
    signUp: SignUp,
    signIn: SignIn
  },
  {
    initialRouteName: 'signIn'
  }
);

/**
 * @property {boolean} - A boolean value indicating whether the current user is logged in or not.
 */

class AuthNav extends Component {
  render() {
    return this.props.isLoggedIn ? <TabNav /> : <SignNav />;
  }
}

export default connect(state => state.auth)(AuthNav);
