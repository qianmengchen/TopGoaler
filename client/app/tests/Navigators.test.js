import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
require('isomorphic-fetch');

import AuthNav from '../components/Navigators/AuthNav';
// import TabNav from '../components/Navigators/TabNav';

const middlewares = [thunkMiddleware]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialTaskState = {
  tasks: ['do frontend', 'do backend', 'do MVP ftw'],
  isFetching: false,
  currentTaskPage: 'ChannelListPage'
};

describe('Testing AuthNav', () => {
  const wrapper = shallow(<AuthNav />, {
    context: { store: mockStore(initialTaskState) }
  });
  const render = wrapper.dive();

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });

  it('renders as expected when logged in', () => {
    const wrapper = shallow(<AuthNav isLoggedIn={true} />, {
      context: { store: mockStore(initialTaskState) }
    });
    const render = wrapper.dive();
    expect(render).toMatchSnapshot();
  });

  // it('has correct icons; needs testing', () => {
  //   // const navigation = {
  //   //   state: {
  //   //     index: 1,
  //   //     routes: [{ routeName: 'Channel', key: 'myroute-123' }]
  //   //   }
  //   // };

  //   const wrapper = shallow(<TabNav routeName={'Channel'} />, {
  //     context: { store: mockStore(initialTaskState) }
  //   });
  //   const render = wrapper.dive();
  //   expect(render).toMatchSnapshot();
  // });
});
