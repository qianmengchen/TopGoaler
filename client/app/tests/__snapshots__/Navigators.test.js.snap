// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AuthNav renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthNav
    dispatch={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;

exports[`Testing AuthNav renders as expected when logged in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthNav
    dispatch={[Function]}
    isLoggedIn={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
